src/main.ts

```diff
@@ -321,9 +321,15 @@ export default class LatexSuitePlugin extends Plugin {
 
 		// Check whether within "\text{}"
 		let withinMath = false;
-		if (withinEquation) withinMath = !isInsideEnvironment(view, pos, {openSymbol: "\\text{", closeSymbol: "}"});
-
+		if (withinEquation) {
+			if (isInsideEnvironment(view, pos, {openSymbol: "\\text{", closeSymbol: "}"}) || isInsideEnvironment(view, pos, {openSymbol: "\\tag{", closeSymbol: "}"})) {
+				withinMath = false;
+			} else {
+				withinMath = true;
+			}
+		} 
 
+		
```

Comment:
```suggestion
		if (withinEquation) withinMath = !(isInsideEnvironment(view, pos, {openSymbol: "\\text{", closeSymbol: "}"}) || isInsideEnvironment(view, pos, {openSymbol: "\\tag{", closeSymbol: "}"}));
```

I think this is cleaner?